jira.go

```diff
@@ -267,9 +289,72 @@ func (c *Client) NewMultiPartRequest(method, urlStr string, buf *bytes.Buffer) (
 	return c.NewMultiPartRequestWithContext(context.Background(), method, urlStr, buf)
 }
 
+func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
+	return c.doWithDeadline(req, v)
+}
+
+func (c *Client) doWithDeadline(req *http.Request, v interface{}) (*Response, error) {
+	ctx := req.Context()
+	if failFast(ctx) {
+		return c.do(req, v)
+	}
+
+	deadline, deadlineIsSet := ctx.Deadline()
+	if !deadlineIsSet {
+		deadline = time.Now().Add(retryIntervalDefault)
+		dctx, cancel := context.WithDeadline(ctx, deadline)
+		defer cancel()
+		ctx = dctx
+		req = req.Clone(ctx)
```

Comment:
```suggestion
		req = req.WithContext(ctx)
```
or is there a reason to Clone it specifically?

Reply:
doc says
> // To change the context of a request, such as an incoming request you
// want to modify before sending back out, use Request.Clone. Between
// those two uses, it's rare to need WithContext.

am I missing something?

Reply:
Right, but "sending back out" means that you received request from somewhere else (e.g. it's running already). In your case it's created by someone and you merely modify it before sending. So it's more in line with `NewRequest` followed by `WithContext`.

Reply:
Done.